pytorch_lightning/core/optimizer.py

```diff
@@ -164,17 +159,16 @@ def closure_dis():
             profiler_action = "optimizer_step_with_closure"
         profiler_action += f"_{self._optimizer_idx}"
 
-        trainer = self._trainer
-        assert trainer is not None
-        with trainer.profiler.profile(profiler_action):
-            trainer.strategy.optimizer_step(self._optimizer, self._optimizer_idx, closure, **kwargs)
+        assert self._strategy is not None
+        assert self._strategy.lightning_module is not None
+        with self._strategy.lightning_module.trainer.profiler.profile(profiler_action):
```

Comment:
This is very ugly but the path of least friction at the moment.

I guess this profiler call will need to be removed. It could only be moved to the loops, but they already profile it through `call_hook`.

Means we will lose the closure suffix.

Left for a follow-up

Reply:
This was there to support manual optimization with profiling.